### PR TITLE
(chore): Upgrade go version

### PR DESCRIPTION
**Pull request by A-440Hz**
Go version 1.19.8 contains a security fix

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Canonical reference for changes, improvements, and bugfixes for Boundary.
 
-## Next
+## 0.12.2 (2023/04/04)
```
Comment (louisruch) [inline CHANGELOG.md:5]: Fly by comment: I believe this should actually be `0.12.1` and we should add a new item for `0.12.2` and include a similar changelog entry as https://github.com/hashicorp/boundary/blob/main/CHANGELOG.md#security

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Please make the changelog updates Louis detailed, seems like we missed it last time.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,15 @@
...
+
+### Security
+
+* Boundary now uses Go 1.19.8 to address security vulnerability (CVE-2023-24536) See the
```
Comment (johanbrandhorst) [inline CHANGELOG.md:9]:
```suggestion
* Boundary now uses Go 1.19.8 to address CVE-2023-24536. See the
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,15 @@
 
 Canonical reference for changes, improvements, and bugfixes for Boundary.
 
-## Next
+## 0.12.2 (2023/04/04)
```
Comment (johanbrandhorst) [inline CHANGELOG.md:5]: I think we probably want to add a ## Next section here as well, for next time.

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]

**Review by johanbrandhorst — APPROVED** [review]
LGTM

**Pull request merged**